app/services/bloom_trade_client/exchange_rates/sync.rb

```diff
@@ -2,20 +2,36 @@ module BloomTradeClient
   module ExchangeRates
     class Sync
 
-      def self.call(data, _)
+      def self.call(data, _, headers)
         exchange_rate = BloomTradeClient::ExchangeRate.where(
-          base_currency: data['base_currency'],
-          counter_currency: data['counter_currency']
+          base_currency: data["base_currency"],
+          counter_currency: data["counter_currency"],
+          jwt_hash: get_jwt_from(headers)
         ).first_or_initialize
 
         exchange_rate.update_attributes!(
-          buy: data['buy'],
-          sell: data['sell'],
-          mid: data['mid'],
-          expires_at: data['expires_at'],
+          buy: data["buy"],
+          sell: data["sell"],
+          mid: data["mid"],
+          expires_at: data["expires_at"],
         )
+      end
+
+      def self.get_jwt_from(headers)
+        return unless headers
+
+        auth = headers["Authorization"]
+        return unless auth
 
+        pattern = /^Bearer /
+
+        if auth && auth&.match(pattern)
```

Comment:
I think you can just have this right?

```suggestion
        if auth&.match(pattern)
```

Reply:
Got it!